test/unit/query-interface/insert.test.ts

```diff
@@ -0,0 +1,103 @@
+import { DataTypes, literal } from '@sequelize/core';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { expectsql, sequelize } from '../../support';
+
+describe('QueryInterface#insert', () => {
+  const User = sequelize.define('User', {
+    firstName: DataTypes.STRING,
+  }, { timestamps: false });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  // you'll find more replacement tests in query-generator tests
+  it('does not parse replacements outside of raw sql', async () => {
+    const stub = sinon.stub(sequelize, 'queryRaw');
+
+    await sequelize.getQueryInterface().insert(null, User.tableName, {
+      firstName: 'Zoe',
+    }, {
+      returning: [':data'],
+      replacements: {
+        data: 'abc',
+      },
+    });
+
+    expect(stub.callCount).to.eq(1);
+    const firstCall = stub.getCall(0);
+    expectsql(firstCall.args[0] as string, {
+      postgres: `INSERT INTO "Users" ("firstName") VALUES ($sequelize_1) RETURNING ":data";`,
+      default: 'INSERT INTO [Users] ([firstName]) VALUES ($sequelize_1);',
+      mssql: `INSERT INTO [Users] ([firstName]) OUTPUT INSERTED.[:data] VALUES ($sequelize_1);`,
+      db2: `SELECT * FROM FINAL TABLE (INSERT INTO "Users" ("firstName") VALUES ($sequelize_1));`,
+      ibmi: `SELECT * FROM FINAL TABLE (INSERT INTO "Users" ("firstName") VALUES ($sequelize_1))`,
+    });
+    expect(firstCall.args[1]?.bind).to.deep.eq({
+      sequelize_1: 'Zoe',
+    });
+  });
+
+  it('throws if a bind parameter name starts with the reserved "sequelize_" prefix', async () => {
```

Comment:
Now that I think about this, why is this test only present in some files?

Reply:
They're the only methods reserving some bind parameter names, because they're the only methods that add extra bind parameters

Reply:
Only insert, update, and bulkUpdate add extra bind parameters in their SQL, so they're the only 3 methods that reserve bind parameters